tests/test_pulp_util.py

```diff
@@ -197,3 +197,53 @@ def test_get_pulp_instance():
     _, workflow = prepare(testfile)
     handler = PulpHandler(workflow, pulp_registry_name, log)
     assert handler.get_pulp_instance() == pulp_registry_name
+
+
+@pytest.mark.skipif(dockpulp is None,
+                    reason='dockpulp module not available')
+@pytest.mark.parametrize(("auto_publish"), [
+    (True),
+    (False),
+])
+def test_ensure_repos(auto_publish):
+    mock_get_data = [
+      {
+        "id": "redhat-myproject-hello-world",
+        "scratchpad": {
+          "tags": [
+            {
+              "image_id": "81c26020120f56f58217eefdbddaad37e0c47d1404e3323c6a7d2648325b08f1",
+              "tag": "none-80435-20170705193414"
+            }
+          ]
+        },
+        "display_name": "redhat-myproject-hello-world",
+        "description": "No description",
+        "distributors": [
+          {
+            "repo_id": "redhat-myproject-hello-world",
+            "auto_publish": auto_publish,
+          }
+        ]
+      }
+    ]
+
+    log = logging.getLogger("tests.test_pulp_util")
+    pulp_registry_name = 'registry.example.com'
+    testfile = 'foo'
+
+    _, workflow = prepare(testfile)
+    handler = PulpHandler(workflow, pulp_registry_name, log)
+    handler.create_dockpulp_and_repos([])
+
+    (flexmock(dockpulp.Pulp)
+     .should_receive('getRepos')
+     .with_args(['redhat-myproject-hello-world'], fields=['id'], distributors=True)
```

Comment:
This is fine when dockpulp is new enough, but I think we also need a separate test in which this call raises `TypeError`. It's fine for that case not to try to update the auto_publish setting (it won't know it needs to).